### PR TITLE
Std

### PR DESCRIPTION
**Pull request by drewtu2**
Wrote Standard deviation function and tests.

### CODE REVIEW

**Comment by yiskylee** [comment]
Hi @drewtu2 , this is not up to date, could you do a pull?

**Comment by drewtu2** [comment]
Conflicts resolved

**Comment by yiskylee** [comment]
1. Please fix the doxygen of StandardDeviation as it still says it is calculating std of a given vector while you are doing this for a matrix. Also make sure to describe what the axis is.
2. returnValue is not a good name regarding it is not really self-explanatory and it doesn't follow naming styles.
3. Spaces should be inserted to things like 1.0/num_rows
4. sqrt() would be more understandable than pow(.5)
5. In line 641, you are printing vector b.

File: cpp/include/cpu_operations.h [hunk 1]
```diff
@@ -616,11 +616,17 @@ class CpuOperations {
...
         std::ptr_fun(util::reciprocal<T>)).matrix().asDiagonal();
   }
-  /// Calculates the standard deviation of a given vector
+  /// Calculates the standard deviation of a given matrix
   ///
   /// \param a
-  /// Input vector
+  /// Input matrix
+  ///
+  /// \param b
```
Comment (yiskylee) [inline cpp/include/cpu_operations.h]: The parameter is called axis, not b

File: cpp/include/cpu_operations.h [hunk 2]
```diff
@@ -616,11 +616,17 @@ class CpuOperations {
...
+  /// The axis for the standard deviation to be calculated along. If axis is 1
+  /// will return the standard deviation calculated by column. If axis is 0,
+  /// will return the standard deviation calcualted by row.
```
Comment (yiskylee) [inline cpp/include/cpu_operations.h]: calculated

**Pull request merged**